### PR TITLE
Complete algorithm challenges

### PR DESCRIPTION
**Pull request by adeolaidowu**
Kindly review my solution to the algorithm challenges

### CODE REVIEW

File: src/binary-reversal/index.js [hunk 1]
```diff
@@ -3,6 +3,13 @@
  *
  *  * @param {string} value
  */
-function binaryReversal(value) {}
+function binaryReversal(value) {
+  const binVal = Number(value).toString(2);
+  const binValArr = binVal.split("");
+  while (binValArr.length < 8) {
+    binValArr.unshift("0");
+  }
```
Comment (bondz) [inline src/binary-reversal/index.js:11]: `binValArr.padStart(8, 0)`

File: src/list-sorting/index.js [hunk 2]
```diff
@@ -1,3 +1,25 @@
-function listSorting(needle, haystack) {}
+function listSorting(needle, haystack) {
+  const arrLength = haystack.length;
+  const answersArray = [];
+  if (typeof haystack[0] == "object") {
```
Comment (bondz) [inline src/list-sorting/index.js:4]: Prefer `Array.isArray(...)` for checking arrays.

File: src/list-sorting/index.js [hunk 3]
```diff
@@ -1,3 +1,25 @@
...
+      return answersArray[0];
+    } else {
+      return answersArray[answersArray.length - 1];
+    }
+  } else {
+    if (haystack.lastIndexOf(needle) !== -1) {
+      return haystack.lastIndexOf(needle);
+    } else {
+      return -1;
+    }
```
Comment (bondz) [inline src/list-sorting/index.js:21]: Missed an opportunity to use `Early Return` here.

File: yarn.lock [hunk 4]
```diff
@@ -0,0 +1,3544 @@
+# THIS IS AN AUTOGENERATED FILE. DO NOT EDIT THIS FILE DIRECTLY.
```
Comment (bondz) [inline yarn.lock:1]: The project uses npm, try to not add additional lockfiles to the repo. One of em could get out of date, atleast until npm7 is released.

**Comment by adeolaidowu** [comment]
Thanks a lot for the insightful feedback.

**Pull request merged**